### PR TITLE
Removed obsolete properties, RecoveryModeSimple and EnvironmentName

### PR DESCRIPTION
**Pull request by erikbra**
(Completes #293 in a better way).

### CODE REVIEW

**Comment by ferventcoder** [comment]
Typically you don't merge your own and you should wait a certain amount of time for input before merging your own

**Comment by ferventcoder** [comment]
I still have an open question on environment names

File: product/roundhouse/infrastructure.app/ApplicationConfiguraton.cs [hunk 1]
```diff
@@ -124,10 +124,7 @@ public static void set_defaults_if_properties_are_not_set(ConfigurationPropertyH
             }
             if (string.IsNullOrEmpty(configuration_property_holder.EnvironmentNames))
             {
-                if (!string.IsNullOrEmpty(configuration_property_holder.EnvironmentName))
-                    configuration_property_holder.EnvironmentNames = configuration_property_holder.EnvironmentName;
-                else
-                    configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;
+                configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;
```
Comment (ferventcoder) [inline product/roundhouse/infrastructure.app/ApplicationConfiguraton.cs]: This change is not quite correct. Still need to be able to pass the environment name from the CLI

Comment (BiggerNoise) [reply product/roundhouse/infrastructure.app/ApplicationConfiguraton.cs]: The configuration_property_holder is only set from environment "names" (in command line, where it's just a synonym). Is this something that needs to be available at the object for users of the RH library?

**Comment by erikbra** [comment]
Yes, sorry. I just thought the merge of the previous one was a b8t
premature, and that this was "part of it", but I definitely see that I
should have kept the command line switch. I just wanted to get rid of all
the warnings, but I was a bit too quick. I'll revert later today, keeping
the EnvironmentName. You are fine with killing off the RecoveryModeSimple,
and just support the RecoveryMode=Simple?

**Comment by erikbra** [comment]
Please see new pull request #296. Learning from my mistakes, I will not merge my own PR minutes after I made it :) So, when the sun rises in the west of the world, would @BiggerNoise  or @ferventcoder pleas take a look?

**Comment by BiggerNoise** [comment]
Thanks for that, I did take a look. I want to better understand the library use case. If the singular 'environment' is just an implementation detail, then we really don't need it.

You're probably right that I merged #293 a little too quickly. Somewhere there's a balance between immediate merging and sending it off to the wine cellar; I'll find it someday...

**Comment by erikbra** [comment]
I can see that. And it's definitely better to merge a bit trigger-happy than to wait 4 years :)
I'm not very used to working with Pull Requests, I see them as a form of communication too, not only a "Request to pull into master", so please tell me if I file PRs too quickly. I might of course just file an issue, and point to my fork of RoundhouseE, and discuss the change before I make a PR. Or is there a way to label a PR with "please don't merge yet - I only want to discuss the change a bit"?

**Comment by ferventcoder** [comment]
Add this to the title in the pr `{WIP}` (work in progress) as in `{WIP}(GH-###) summary`

**Comment by erikbra** [comment]
Like so? https://github.com/chucknorris/roundhouse/pull/294

**Pull request merged**